blockstack_client/backend/drivers/ipfs.py

```diff
@@ -61,38 +60,33 @@ def write_chunk( chunk_path, chunk_buf, is_mutable ):
     Return True on success 
     Return False on error, and log an exception
     """
-    
-    api = ipfsapi.connect(IPFS_SERVER, IPFS_PORT)
 
     rc = True
-    begin = None
-    end = None
-    size = None
+
+    if IPFS_COMPRESS:            
+      compressed_data = compress_chunk( chunk_buf )
+    else:
+      compressed_data = chunk_buf
+
     try:
-        if IPFS_COMPRESS:            
-            compressed_data = compress_chunk( chunk_buf )
-        else:
-            compressed_data = chunk_buf
-
-        size = len(compressed_data)
-
-        begin = time.time()
-        h = api.add_str( compressed_data, decoder='json' )
-        if is_mutable:
-          if os.environ.get("BLOCKSTACK_TEST") == "1":
-            log.debug("[DEBUG] ipfs.publish_name %s: %s" % (chunk_path, chunk_buf))
-          r = api.name_publish( h, opts = {'key': 'mutable-{}'.format(chunk_path.replace( "/", r"\x2f" ))})
-          if os.environ.get("BLOCKSTACK_TEST") == "1":
-            log.debug("[DEBUG] ipfs.publish_name returned " % (r))
-        end = time.time()
-        
+      h = ipfs_api.add_str( compressed_data, decoder='json' )        
```

Comment:
I can confirm by looking at the output that this was indeed the case: the script processes the data and stores it as binary data in IPFS, and then retrieves it and decompresses correctly matching the input. I've removed the decoder option as per @sunnya97 suggestion.